src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java

```diff
@@ -344,4 +344,39 @@ void shouldReturnSuccess_Empty_ServiceID() {
         }
     }
 
+    private void validateSuccessRetriveCaseFlagResponse(Response response) {
+        if (OK.value() == response.getStatusCode()) {
+            var caseFlags = response.getBody().as(CaseFlag.class);
+            assertNotNull(caseFlags);
+            assertThat(caseFlags.getFlags()).hasSizeGreaterThanOrEqualTo(1);
+        } else {
+            assertEquals(NOT_FOUND.value(), response.getStatusCode());
+        }
+    }
+
+    @Test
+    @ToggleEnable(mapKey = MAP_KEY_CASE_FLAGS, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    void shouldReturnBadRequestForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsEmpty() {
+        final var response = (ErrorResponse)
+            commonDataApiClient.retrieveCaseFlagsByServiceId(
+                HttpStatus.BAD_REQUEST,
+                "service-id=XXXX?available-external-flag="
+            );
+        assertEquals(response.getErrorCode(),400);
+        assertEquals(response.getErrorDescription(),"Allowed values are Y or N");
+    }
+
+    @Test
+    @ToggleEnable(mapKey = MAP_KEY_CASE_FLAGS, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    void shouldReturnNotFoundForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagAndDefaultStatus() {
+        final var response = (ErrorResponse)
+            commonDataApiClient.retrieveCaseFlagsByServiceId(
+                NOT_FOUND,
+                "service-id=XBBB?available-external-flag=Y&default-status=true"
```

Comment:
endpoint not expecting default-status as query param , what's use of this test case ?